src/App.tsx

```diff
@@ -1,25 +1,40 @@
-import React from 'react';
+import React, { ChangeEvent, useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+import { Pagination } from './components/Pagination';
 
 export const App: React.FC = () => {
+  const totalAmount = 42;
+  const items = getNumbers(1, 42).map(n => `Item ${n}`);
+  const [value, setValue] = useState(5);
+  const [sortPage, setSortPage] = useState(1);
+
+  const handlePageChange = (page: number) => {
+    setSortPage(page);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {sortPage} (items {(sortPage - 1) * value + 1} -{' '}
+        {Math.min(sortPage * value, totalAmount)} of {totalAmount})
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
+            className="form-control"
+            value={value}
+            onChange={(e: ChangeEvent<HTMLSelectElement>) => {
+              setValue(Number(e.target.value));
+              setSortPage(1);
+            }}
```

Comment:
Pls move the function inside `onChange` handler to a separate const and use as a callback here